app/client/index.tsx

```diff
@@ -49,21 +61,40 @@ const sidebarRoutes = [
     }
 ];
 
-class Root extends React.Component<{}, {}> {
+class Root extends React.Component<{}, {signedIn: boolean}> {
+    private static didAttachListener: boolean = false;
+
     constructor() {
         super();
+        this.state = {
+            signedIn: CONFIG.getLongName().caseOf({
+                just: _ => true,
+                nothing: () => false
+            })
+        };
+
+        if (!Root.didAttachListener) {
+            CONFIG.addLongNameChangeListener((longNameOpt: Maybe<string>) => {
+                longNameOpt.caseOf({
+                    just: (_: string) => this.setState({ signedIn: true }),
+                    nothing: () => this.setState({ signedIn: false })
+                });
+            });
+            Root.didAttachListener = true;
+        }
     }
 
     render() {
+        const showSidebarRoutes = sidebarRoutes.filter((s: SidebarRoute) => this.state.signedIn || s.title !== 'Me');
```

Comment:
Maybe we should also prevent people from reaching the Upload page as well if they aren't signed in.